extension/src/createProject/SpecifySourcePackageNameStep.ts

```diff
@@ -39,15 +45,26 @@ export class SpecifySourcePackageNameStep implements IProjectCreationStep {
                 );
             }
             disposables.push(
+                inputBox.onDidChangeValue(async () => {
+                    const normalizedName = await getNormalizedPackageNameTrigger(inputBox.value);
+                    if (!normalizedName) {
+                        return;
+                    } else if (normalizedName !== inputBox.value) {
+                        this.setInputInvalid(inputBox, normalizedName as string);
+                    } else {
+                        this.setInputValid(inputBox);
+                    }
+                }),
                 inputBox.onDidAccept(async () => {
-                    const normalizedName = await metadata.client.getNormalizedPackageName(inputBox.value);
-                    if (normalizedName === inputBox.value) {
+                    const normalizedName = await getNormalizedPackageNameTrigger(inputBox.value);
```

Comment:
will there be a 500ms delay after I press `enter`?

Reply:
Yes. Do you think it's a bit longer?

Reply:
We don't have to use debounce here, as it's only triggered once by `enter`. Is there any side effect?

Reply:
e.g. `aaabbb`, insert `-` between `aaa` and `bbb` and press `enter` quickly. consider the behavior.

Reply:
if we remove the debounce here, in this case,

- if the input value is invalid, `setInputInvalid()` seems to be called twice and no side effect.
- if the input value is valid, the handler in `onDidAccept` will return first, and when the debounce in `onDidChangeValue` returns, it's possible the **inputBox** has disposed. calling `setInputValid()` might cause an NPE. but when I tried this, I find the `setInputValid()` function can successfully executed after **inputBox** has disposed. No exception found.

Reply:
500ms after pressing `enter` is not responsive. I myself prefer to remove it for better user experience. 

If we can make sure there's no unexpected behavior, let's just remove the debounce here. But it's also okay to keep it for lower risk if you are not confident enough about the behavior.

Reply:
Yeah, I'll remove debounce here and add some null checks (e.g., inputBox?.enabled = false) in case of any unexpected scenario and upstream change.